# Descriptive Statistics

## Week 2 – Day 1

This notebook explores descriptive statistics using a real dataset. It computes measures of central tendency and measures of spread, then interprets the results to understand the distribution of the data.
___

## Why Statistics Comes Before Modeling

Before building machine learning models, it is important to understand the dataset. Descriptive statistics summarize the data by showing its center and spread, helping identify patterns, variability, and possible outliers. This makes it easier to choose appropriate preprocessing and modeling techniques.
___

In [45]:
import numpy as np
import pandas as pd

This code imports the NumPy and Pandas libraries, which are used for numerical computations, data manipulation, and statistical analysis throughout the notebook.

___
## Dataset

The dataset was loaded into a Pandas DataFrame, and a numeric column was selected for statistical analysis.

In [46]:
df=pd.read_csv("students.csv")
df.head()
ages=df['age']
ages.head(5)

0    19
1    20
2    18
3    19
4    60
Name: age, dtype: int64

## Measures of Central Tendency

The mean, median, and mode are computed to describe the center of the data.

- **Mean:** The arithmetic average of all values.
- **Median:** The middle value after sorting the data.
- **Mode:** The most frequently occurring value.

In [47]:
mean_age = np.mean(ages)
median_age = np.median(ages)
mode_age = pd.Series(ages).mode()[0]
print("Mean Age:", mean_age)
print("Median Age:", median_age)
print("Mode Age:", mode_age)

Mean Age: 23.5
Median Age: 19.5
Mode Age: 18


## Measures of Spread

Measures of spread describe how much the values vary around the center.

The following statistics are computed:

- Variance
- Standard Deviation
- Interquartile Range (IQR)

In [48]:
variance_age = np.var(ages)
standard_deviation_age = np.std(ages)

q1=np.percentile(ages, 25)
q3=np.percentile(ages, 75)
iqr_age = q3 - q1

print("Variance Age:", variance_age)
print("Standard Deviation Age:", standard_deviation_age)
print("Q1 Age:", q1)
print("Q3 Age:", q3)
print("Interquartile Range Age:", iqr_age)

Variance Age: 149.65
Standard Deviation Age: 12.23315167894194
Q1 Age: 18.25
Q3 Age: 20.75
Interquartile Range Age: 2.5


# Which better represents the typical value: the mean or the median? Why?

In [49]:
if abs(mean_age - median_age) < 0.2 * standard_deviation_age:
    print("The mean and median are close to each other, indicating a balanced distribution.")
    print("The mean is a good representation of the typical value")
else:
    print("The mean differs noticeably from the meadian")
    print("The meadian is a good representation of the typical value")

The mean differs noticeably from the meadian
The meadian is a good representation of the typical value


## Interpretation

The mean, median, standard deviation, and IQR help describe the center and spread of the data. When the dataset contains an outlier, such as an unusually high value (e.g., 60), the mean becomes larger than the median because it is affected by extreme values. In this case, the median provides a better representation of the typical observation. The standard deviation also increases, indicating greater variability in the data.

On the other hand, when the data does not contain significant outliers and most observations are clustered around similar values (e.g., around 19 years old), the mean and median are close to each other, suggesting that the distribution is approximately symmetric. In such cases, the mean provides a good representation of the typical value. The IQR remains a useful measure of the spread of the middle 50% of the data and is generally less sensitive to outliers than the standard deviation.

____

### Summary

The average age is 23.5 years, while the median age is 19.5 years.

The difference between the mean and median indicates that the dataset contains an outlier (age 60), which pulls the mean upward.

The standard deviation is relatively high, showing noticeable variation in the ages.

The IQR is 3 years, meaning that the middle 50% of the ages are concentrated within a narrow range.

Overall, the median provides a better representation of the typical age because it is less affected by the outlier.